File: src/main/java/com/taranovski/leetcode/solutions/easy/leetcode0007/java/Solution.java
package com.taranovski.leetcode.solutions.easy.leetcode0007.java;

/**
 * Created by Alyx on 29.09.2018.
 */
public class Solution {
    public int reverse(int x) {

        if (x == 0) {
            return 0;
        }

        boolean isNegative = x < 0;
        long number = isNegative ? (- (long) x) : ((long) x);

        StringBuilder stringBuilder = new StringBuilder();

        long remainder1;
        do {
            remainder1 = number % 10;
            if (remainder1 == 0) {
                number = number / 10;
            }
        } while (remainder1 == 0);

        while (number > 0) {
            long remainder = number % 10;
            number = number / 10;

            stringBuilder.append(remainder);
        }

        final long parseLong = Long.parseLong(stringBuilder.toString());

        if (isNegative && -parseLong < Integer.MIN_VALUE) {
            return 0;
        }
        if (!isNegative && parseLong > Integer.MAX_VALUE) {
            return 0;
        }
        return isNegative ? (- (int) parseLong) : ((int) parseLong);
    }

}

File: src/main/java/com/taranovski/leetcode/solutions/easy/leetcode0007/kotlin/Solution.kt
package com.taranovski.leetcode.solutions.easy.leetcode0007.kotlin

/**
 * Created by Alyx on 29.09.2018.
 */
class Solution {
    fun reverse(x: Int): Int {

        if (x == 0) {
            return 0
        }

        val isNegative = x < 0
        var number = if (isNegative) -x.toLong() else x.toLong()

        val stringBuilder = StringBuilder()

        var remainder1: Long
        do {
            remainder1 = number % 10
            if (remainder1 == 0L) {
                number /= 10
            }
        } while (remainder1 == 0L)

        while (number > 0) {
            val remainder = number % 10
            number /= 10

            stringBuilder.append(remainder)
        }

        val parseLong = java.lang.Long.parseLong(stringBuilder.toString())

        if (isNegative && -parseLong < Integer.MIN_VALUE) {
            return 0
        }
        if (!isNegative && parseLong > Integer.MAX_VALUE) {
            return 0
        }
        return if (isNegative) -parseLong.toInt() else parseLong.toInt()
    }

}
